### ISSUE TITLE
Bug? >= 3rd repetition in custom shortcut config can get into weird state.

### ISSUE THREAD

**Issue by gimbo** [rxhanson/RectanglePro-Community#341]
I've just discovered RectanglePro and I'm loving it. But I noticed something weird while setting up a custom shortcut with repetition...

## Short version

If I create a 3rd, 4th, 5th, ... repetition by selecting dimensions of an existing window (rather than a blank "New Repetition") then:

* I can't delete that repetition.
* I can't add subsequent repetitions.
* The repetition has a name not a number.
* The repetition's JSON has a `display` field which would otherwise not be there and seems superfluous.

## Long version

Suppose I'm trying to create a Custom Shortcut which basically cycles through "Left Half", "First Third", and "First Two Thirds"; the first two repetitions might look like this:

<img width="399" alt="Screenshot 2023-08-29 at 18 51 45" src="https://github.com/rxhanson/RectanglePro-Community/assets/133770/e854773a-2280-4dc1-b3a8-19edb1506594">

Here's the corresponding JSON from exporting the config; also nice and simple:

```json
{
    'name': 'cycle-left',
    'id': 371,
    'w': '0.5',
    'h': '1.0',
    'loc': 3,
    'children': [
        {
            'id': 387,
            'w': '0.333',
            'h': '1.0',
            'loc': 3
        }
    ]
}
```

Now, if I add the third step by hovering over step 2 and hitting `+`, I see a menu something like this...

<img width="308" alt="Screenshot 2023-08-29 at 18 54 56" src="https://github.com/rxhanson/RectanglePro-Community/assets/133770/7503d819-d6d9-441f-9f28-4ddaafbc1c68">

If I choose "New Repetition" everything will be fine: I get a nice "3rd execution" window just like the "2nd execution" window corresponding to step 2. I might fill it out like so:

<img width="299" alt="Screenshot 2023-08-29 at 18 56 04" src="https://github.com/rxhanson/RectanglePro-Community/assets/133770/19597aae-f409-472e-957e-20c9dbc6ff31">

The JSON is the same except for one more `children` element:

```json
...
        {
            'id': 402,
            'w': '0.666',
            'h': '1.0',
            'loc': 3
        }
...
}
```

**However** —  now comes the problem.

If instead I create the 3rd step by picking window dimensions from the menu shown above (i.e. NOT "New Repetition"), I get something like this:

<img width="944" alt="Screenshot 2023-08-29 at 18 59 43" src="https://github.com/rxhanson/RectanglePro-Community/assets/133770/b1e510c7-4aa4-4cd2-99ea-fd1ce6448797">

This is different; it's more like I'm creating a named step than a numbered one. So e.g. I can end up with this:

<img width="944" alt="Screenshot 2023-08-29 at 19 00 40" src="https://github.com/rxhanson/RectanglePro-Community/assets/133770/04c80e55-e61c-4648-9b5f-5e08f94e25f1">

It happens that as a shortcut this still works. Unfortunately, it's also broken:

1. Clicking on `-` doesn't delete the repetition.
2. Clicking on `+` doesn't add another repetition.

It's also got different JSON: the 3rd step now includes `'name': 'FOO BAR BAZ'` (which is fine) and `'display': 0` (which is unexpected and strange).

Finally, I note that this _does not_ happen if I create my step 2 by picking window dimensions: that works fine, giving me a numbered step just like if I'd clicked "New Repetition".

As such, I'm guessing this is a bug.

**Comment by rxhanson** [comment]
Thanks for the detailed report. Certainly a bug - I'll slate this one for the next release.

**Comment by rxhanson** [comment]
This one should now be fixed in 3.0.8, available within the app or at https://rectangleapp.com/pro
